src/frontend/src/modals/IOModal/components/chatView/chat-view.tsx

```diff
@@ -142,6 +143,20 @@ export default function ChatView({
 
   const flowRunningSkeletonMemo = useMemo(() => <FlowRunningSqueleton />, []);
 
+  const memoChatHistory = useMemo(() => {
+    return chatHistory?.map((chat, index) => (
+      <ChatMessage
+        setLockChat={setLockChat}
+        lockChat={lockChat}
+        chat={chat}
+        lastMessage={chatHistory.length - 1 === index ? true : false}
+        key={`${chat.id}-${index}`}
+        updateChat={updateChat}
+        closeChat={closeChat}
+      />
+    ));
+  }, [chatHistory]);
+
   return (
```

Comment:
What if you memoize each message instead? That way it keeps the message stable and just re-renders the ones that get updated instead of the entire ChatHistory